packages/docusaurus/lib/client/clientEntry.js

```diff
@@ -11,10 +11,12 @@ import {BrowserRouter} from 'react-router-dom';
 
 import App from './App';
 import preload from './preload';
+import docusaurus from './docusaurus';
 import routes from '@generated/routes'; // eslint-disable-line
 
 // Client-side render (e.g: running in browser) to become single-page application (SPA).
 if (typeof window !== 'undefined' && typeof document !== 'undefined') {
+  window.__docusaurus = docusaurus;
```

Comment:
There will be a cyclic dependency if downstream try to import it. Mainly due to `routes`

It is also easier to do `__docusaurus` instead of `import + then call the function`.
This make it possible to have one source of truth
Algolia is doing this as well, the api runner is actually in `window.__algolia`
Gatsby prefetching is done like this too. `window.__loader`

Their prefetch is
```js
___loader.enqueue(parsePath(this.props.to).pathname)
```

https://github.com/gatsbyjs/gatsby/blob/master/packages/gatsby-link/src/index.js#L61